Refactor project structure, separate concerns, and enhance usage documentation

x0rw opened the pull request:
Hey :wave: , love this project.
- Moved internal lock and helper functions out of `lib.rs` to improve clarity and separation of concerns.
- Isolated error types into a dedicated `errors.rs `module for better error handling and modularity.
- Restructured the project layout by moving integration tests to the` /tests `directory.

Updated README.md with comprehensive usage examples:
- Environment-based configuration
- Manual client configuration
- Service registration and deregistration

i'm working on adding [acl](https://developer.hashicorp.com/consul/docs/security/acl) feature on my branch after this is pushed, i will open a pr for it too.

# Checks
Please check these off before promoting the pull request to non-draft status.
- [x] All CI checks are green.
- [x] I have reviewed the proposed changes myself.

x0rw commented:
I have read the CLA Document and I hereby sign the CLA

kushudai commented on Cargo.toml:
```diff
@@ -30,4 +30,8 @@ serde_json = "1.0"
 slog-scope = "4"
 smart-default = "0.7"
 tokio = { version = "1", features = ["full"] }
+tokio-test = "0.4"  # Explicitly add if missing
```
Could we move this to dev dependencies please?

kushudai commented on src/lock.rs:
```diff
@@ -0,0 +1,118 @@
+/// The lifetime of this object defines the validity of the lock against consul.
+/// When the object is dropped, the lock is attempted to be released for the next consumer.
+#[derive(Clone, Debug)]
+pub struct Lock<'a> {
```
Unless I missed it, this needs a `pub use` to be visible to consumers of this crate?

kushudai reviewed (commented):
Hi @x0rw, thank you for this PR!

My main concern here is primarily the breaking changes from types moving to be within modules.

Any thoughts on re-exporting them from lib.rs?

x0rw commented:
This is the report generated by `cargo public-api diff`. There are no changes or removals to the public API now.
Additionally, quick_error!() is unmaintained(i [checked](https://crates.io/crates/quick_error), the last update was 4 years ago) and no longer considered ergonomic, lacking proper visibility control. It would be beneficial to migrate to `thiserror` for long-term maintainability.
```
Removed items from the public API
=================================
(none)

Changed items in the public API
===============================
(none)

Added items to the public API
=============================
+impl rs_consul::Consul
```

kushudai commented:
> This is the report generated by `cargo public-api diff`. There are no changes or removals to the public API now. Additionally, quick_error!() is unmaintained(i [checked](https://crates.io/crates/quick_error), the last update was 4 years ago) and no longer considered ergonomic, lacking proper visibility control. It would be beneficial to migrate to `thiserror` for long-term maintainability.
> 
> ```
> Removed items from the public API
> =================================
> (none)
> 
> Changed items in the public API
> ===============================
> (none)
> 
> Added items to the public API
> =============================
> +impl rs_consul::Consul
> ```

Thank you, you're right. This crate has not seen as much love as I would like it to.
You're welcome to fix that in this PR but I don't want ask more of you or throw scope creep on this change.

I'm happy to pick that up separately.

x0rw commented:
It's not an issue at all, I'd be happy to look into that.

kushudai commented on Cargo.toml:
```diff
@@ -24,10 +24,13 @@ hyper = { version = "1", features = ["full"] }
 serde_json = "1.0"
 slog-scope = "4"
 smart-default = "0.7"
+thiserror = "2.0.12"
```
Nit: We try to limit to the most significant part of the version as it prevents confusion for newer rust developers who may think this is being pinned to a particular version
```suggestion
thiserror = "2"
```

kushudai reviewed (approved):
Feel free to mark this as PR ready when you feel comfortable about merging this.
Again, thank you for the work!

x0rw commented:
It's ready for merging now, i will work on a feature next week.
